store/sqlstore/post_store.go

```diff
@@ -1206,6 +1206,16 @@ func (s *SqlPostStore) search(teamId string, userId string, params *model.Search
 		}
 	}
 
+	terms, containsEmails := wrapEmails(terms)
+	if containsEmails {
+		// Find and remove "@"
+		for i, v := range specialSearchChar {
+			if v == "@" {
+				specialSearchChar = append(specialSearchChar[:i], specialSearchChar[i+1:]...)
```

Comment:
This will also remove that character for other terms where we may want to replace it, non-email terms for example